.github/workflows/ci-build.yml

```diff
@@ -8,29 +8,37 @@ on:
 
 env:
   CARGO_TERM_COLOR: always
+  RUSTFLAGS: "-D warnings"
 
 jobs:
   build:
+    strategy:
+      matrix:
+        features: [
+          "--all-features",
+          "--features=default",
+          "--features=test-vector",
+          "--features=multithreaded",
+          "--features=test-vector,multithreaded",
+          "--features=prio2",
+          "--features=prio2,test-vector",
+          "--features=prio2,multithreaded",
+          "--features=prio2,test-vector,multithreaded",
```

Comment:
This looks like a combinatorial explosion. If we end up needing more feature flags, will we have to continue testing all possible combinations? Anything we can do to avoid this complexity?

Reply:
cc/ @tgeoghegan

Reply:
IMO we don't have a choice but to test all the feature flag combinations we intend to support. We can try our best to make sure that one feature doesn't break another, but we can't have confidence in that unless we actually build and test the feature flag combo. I think this means we have to be careful about adding features. The alternative would be to carefully document what feature combinations we do test and make it clear to clients that they are responsible for testing other feature combos.

As for the length of this list: there's probably a more clever way to write a [workflow matrix strategy](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix), but since there are currently only 8 configurations, I find it easier to reason about if we spell it out this way.

Reply:
Ack, in that case it's *great news* that we've lumped "enpa" and "prio2" together!

Reply:
There's a tool available that could help with this, [https://github.com/frewsxcv/cargo-all-features](cargo-all-features). It doesn't do clippy or benchmarks, though.